### PR TITLE
Build and publish tkn image on release 📌

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

- Update release pipeline and tasks to v1beta1
- Build and publish official tkn image on the release.

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

/cc @chmouel @danielhelfand 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Run the code checkers with `make check`
- [ ] Regenerate the manpages, docs and go formatting with `make generated`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```
Release the tkn official image as part of the tkn release process
```

### CODE REVIEW

File: tekton/get-version.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  resources:
+    inputs:
+    - name: source
+      type: git
+  steps:
+  - name: pull
+    image: goreleaser/goreleaser
+    workingDir: /workspace/source
+    script: |
+      git status; git fetch -p --all
```
Comment (chmouel) [inline tekton/get-version.yaml]: `&&` instead of `;` perhaps?

File: tekton/get-version.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  - name: get-versions
+    image: goreleaser/goreleaser # because it has git
+    workingDir: /workspace/source
+    script: |
+      echo -n $(git tag --points-at HEAD) > /tekton/results/version
+      echo -n $(git tag --points-at HEAD | sed 's/^.//' | awk -F '[.]' '{ print $1}') > /tekton/results/major
```
Comment (chmouel) [inline tekton/get-version.yaml]: perhaps shorter `echo -n $(git tag --points-at HEAD | sed 's/^v\([^\.]*\).*/\1/'`) > `

your command (and mine) would get the `0` of `0.8.1`, is that what you want ? don't you want something like 0.8 instead? and `1` being the minor?

Comment (vdemeester) [reply tekton/get-version.yaml]: In the end, in results I want `v0.8.0` (version), `0` (major), `8` (minor), in the result files.

Comment (chmouel) [reply tekton/get-version.yaml]: so you can do : 

major:
> git tag --points-at HEAD | sed 's/^v\([^\.]*\)\.\([^\.]*\).*/\1/'

minor:

> git tag --points-at HEAD | sed 's/^v\([^\.]*\)\.\([^\.]*\).*/\2/'

**Comment by vdemeester** [comment]
/cc @danielhelfand @chmouel @piyush-garg @pradeepitm12 
Does this looks ok ?

**Comment by chmouel** [comment]
looks fine by me but i haven't tested it..

File: tekton/publish.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    description: Path to the Dockerfile to build.
+    default: ./Dockerfile
+  - name: CONTEXT
+    description: The build context used by Kaniko.
+    default: ./
+  - name: EXTRA_ARGS
+    type: array
+  - name: BUILDER_IMAGE
+    description: The image on which builds will run
+    default: gcr.io/kaniko-project/executor:latest
```
Comment (danielhelfand) [inline tekton/publish.yaml:17]: Should we tag to a specific version?

File: tekton/get-version.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: get-versions
+spec:
+  resources:
+    inputs:
+    - name: source
```
Comment (danielhelfand) [inline tekton/get-version.yaml:8]: For consistency with the Pipeline, can this be `source-repo`? That or change this PipelineResource on the Pipeline to `source`?

File: tekton/release-pipeline.yml [hunk 5]
```diff
@@ -8,26 +8,36 @@ spec:
   - name: package
     description: package to release
     default: github.com/tektoncd/cli
-  # FIXME(vdemeester): this is not yet used and need to be fixed upstream
   - name: github-token-secret
     description: name of the secret holding the github-token
     default: bot-token-github
+  - name: github-token-secret-key
+    description: name of the secret holding the github-token
```
Comment (danielhelfand) [inline tekton/release-pipeline.yml:15]:
```suggestion
    description: name of the key for the secret holding the github-token
```

File: tekton/get-version.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+      type: git
+  steps:
+  - name: pull
+    image: goreleaser/goreleaser
+    workingDir: /workspace/source
+    script: |
+      git status && git fetch -p --all
+  - name: get-versions
+    image: goreleaser/goreleaser # because it has git
```
Comment (danielhelfand) [inline tekton/get-version.yaml:17]: Should we tag to a specific version?

File: tekton/publish.yaml (hunk 3 above)
Comment (vdemeester) [reply tekton/publish.yaml:17]: Yeah

File: tekton/get-version.yaml (hunk 4 above)
Comment (vdemeester) [reply tekton/get-version.yaml:8]: Yeah :wink:

File: tekton/get-version.yaml (hunk 6 above)
Comment (vdemeester) [reply tekton/get-version.yaml:17]: indeed !

**Comment by vdemeester** [comment]
@chmouel @danielhelfand updated :wink:

**Pull request merged**